### ISSUE TITLE
Results nnUNet fold 0 12 subject 

### ISSUE THREAD

**Issue by tzebre** [ivadomed/model-spinal-rootlets#5]
## Used data 
Spinal roots are visible only on T2w MRI.
Original data are available here:[ds004507/versions/1.0.1](https://openneuro.org/datasets/ds004507/versions/1.0.1)
Since no dataset with a mask on the spinal root is available, a manual labeling was done for the subjects of this dataset (16 MRI, headNormal and headUp).
I converted it to the [nnUNet V2 standard dataset format](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/dataset_format.md).
The dataset is available at duke/temp/theo_root/Dataset001_rootseg.
Test images are all MRIs from [ds004507/versions/1.0.1](https://openneuro.org/datasets/ds004507/versions/1.0.1) (headDown, headNormal, headUp) -> duke/temp/theo_root/Dataset001_rootseg/imagesTs 

## Command to reproduce 
Conversion from BIDS to nnUNet done with this [notebook](https://github.com/ivadomed/model-spinal-rootlets/blob/sandbox/nnUnet_DS.ipynb). Correspondence between BIDS names and nnUNet are in this csv files:
- [without headDown](https://github.com/ivadomed/model-spinal-rootlets/blob/sandbox/new_name.csv)
- [with headDown](https://github.com/ivadomed/model-spinal-rootlets/blob/sandbox/new_name_withDown.csv)
### a) Verify record integrity 
`nnUNetv2_plan_and_preprocess -d 001 --verify_dataset_integrity`.
### b) Train a 3d_fullres configuration 
Train only one fold (#0)
`CUDA_VISIBLE_DEVICES=1 nnUNetv2_train 001 3d_fullres 0`.

### c) Predict MRI 
`nnUNetv2_predict -i PATH_TO:imagesTs -o PATH_TO:Out_directory -d 001 -c 3d_fullres --save_probabilities -chk checkpoint_best.pth`

### a) Training
After 50 epoch: 
<img src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/d833f105-6e47-4f8c-afdc-220bfd68ef55" width="600">
### b) Inference 

headDown no good result (difficult to see the rootlet in this position)
headNormal/headUp really good results, sometimes better than my manual predictions. 
Example for a subject not in the training data set and never manually labeled sub-010_headNormal (mri_016):
<img width="1021" alt="Capture d'écran 2023-05-29 à 10 57 18" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/00283577-149e-4768-bcea-1414adcbee85">
<img width="1021" alt="Capture d'écran 2023-05-29 à 10 57 26" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/615bbc9c-9566-4900-be4e-1d5090e8444e">
Without touching the contrast, we can think that the prediction is wider than the rootlets. If we modify the contrast a little, it's less the case. 
<img width="995" alt="Capture d'écran 2023-05-29 à 10 57 39" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/b9277fa3-45f5-4a1f-a857-c9136c74f3a5">

Exemple with a subject in the dataset previously labeled sub-002_ses-headNormal (mri_002):
<img width="986" alt="Capture d’écran 2023-05-29 à 11 03 21" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/5cd42bc7-4bf1-4576-98b8-e74d8128328f">
blue = nnUNet predict, red = Manual label 
<img width="929" alt="Capture d’écran 2023-05-29 à 11 03 34" src="https://github.com/ivadomed/model-spinal-rootlets/assets/58122097/ed86fad2-27ec-4f86-991a-cc38dc20b5d7">

## Discussion
The results are really encouraging. 
Need to test on different dataset with different resolutions. 
I've used the prediction to label the full dataset and I'm going to train the 5-fold on these datasets.

**Comment by jcohenadad** [comment]
Very encouraging indeed!!! 

> Without touching the contrast

Can you please clarify what you mean by 'touching the contrast'? What exactly do you do to the image to affect the contrast? and don't you have some sort of batch normalization and data augmentation that affects image characteristics (intensity bias, intensity shift, intensity scaling, gamma, etc.)

**Comment by tzebre** [comment]
> Very encouraging indeed!!! 
> 
> 
> 
> > Without touching the contrast
> 
> 
> 
> Can you please clarify what you mean by 'touching the contrast'? What exactly do you do to the image to affect the contrast? and don't you have some sort of batch normalization and data augmentation that affects image characteristics (intensity bias, intensity shift, intensity scaling, gamma, etc.)

By touching the contrast I just mean changing the contrast and brightness value in fsleyes to increase the rootless visibility. I'm not sure to understand the other questions well, but I didn't change the default preprocessing normalization of nnUNet.

**Comment by sandrinebedard** [comment]
Very encouraging! Have you tried denoising the images bofore training? I think it helps visualize the rootlets , here is an example:

![image](https://github.com/ivadomed/model-spinal-rootlets/assets/71230552/0a66da8e-ca9e-4378-a844-e752f5665f0e)

**Comment by tzebre** [comment]
> Very encouraging! Have you tried denoising the images bofore training? I think it helps visualize the rootlets , here is an example:
> 
> ![image](https://user-images.githubusercontent.com/71230552/244752520-0a66da8e-ca9e-4378-a844-e752f5665f0e.png)

Thank. I've just tried and it helps a lot to clear the images and increase the rootlet visibility.

**Comment by jcohenadad** [comment]
> Thank. I've just tried and it helps a lot to clear the images and increase the rootlet visibility.

Do you mean that it helps the model to improve its performance, or that it helps human (ie: you) to visualise the rootlets for generating the ground truth?  I would be surprised if it was the former, given that our experience with denoising training data show negligible improvement on model performance

**Comment by tzebre** [comment]
> Do you mean that it helps the model to improve its performance, or that it helps human (ie: you) to visualise the rootlets for generating the ground truth? I would be surprised if it was the former, given that our experience with denoising training data show negligible improvement on model performance

Yes, it was not clear, but help creating ground truth. I haven't tried the model with denoised images
